rasterio/tools.py

```diff
@@ -0,0 +1,62 @@
+"""Rasterio tools module
+
+See this RFC about Rasterio tools:
+https://github.com/mapbox/rasterio/issues/1300.
+"""
+
+import json
+
+import rasterio
+from rasterio.features import dataset_features
+
+
+class JSONSequenceTool(object):
+    """Extracts data from a dataset file and saves a JSON sequence
+    """
+
+    def __init__(self, func):
+        """Initialize tool
+
+        Parameters
+        ----------
+        func : callable
+            A function or other callable object that takes a dataset and
+            yields JSON serializable objects.
+        """
+        self.func = func
+
+    def __call__(self, src, dst, src_opts=None, dst_opts=None, config=None):
+        """Execute the tool's primary function
+
+        Parameters
+        ----------
+        src : str or PathLike object
+            A dataset path or URL. Will be opened in "r" mode using
+            src_opts.
+        dst : str or Path-like object
+            A path or or PathLike object. Will be opened in "w" mode.
```

Comment:
We also use `src` and `dst` to refer to dataset objects in a many places, both in rasterio tests and in rio modules. Typing `with rasterio.open(...) as src` is basically muscle memory at this point, for better or worse! Can we use different argument names here to avoid any confusion? (`src_path`?)